uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-balance-tab.component.js

```diff
@@ -37,8 +37,11 @@ define([
      * and watches for events coming from child components.
      */
     (function init () {
-      loadDependencies();
-      watchForEvents();
+      initWatchers();
+
+      loadDependencies().then(function () {
+        vm.loading.component = false;
```

Comment:
@AkA84 and I had that discussion already and agreed to move this to the init function. The reason is that component loading has more to do with the init function than with loading dependencies... unless I change the var name to `vm.loading.dependencies`.